pkg/install/install.go

```diff
@@ -200,5 +200,5 @@ func waitForCRDsEstablished() error {
 
 // waitForPodsReady waits for all pods in the given namespace to be ready.
 func waitForPodsReady(ns string) error {
-	return runCommand(exec.Command("kubectl", "wait", "pod", "--timeout=-1s", "--for=condition=Ready", "-l", "!job-name", "-n", ns))
+	return runCommand(exec.Command("kubectl", "wait", "pod", "--timeout=5m", "--for=condition=Ready", "-l", "!job-name", "-n", ns))
```

Comment:
I don't think we want to hardcode this to 5 minutes, if pods are not ready there is something wrong. like in the issue there is a problem with `ImagePullBackOff` error 

```
knative-serving      activator-5cd9886d74-9qmz2                      0/1     ImagePullBackOff   0          14m
knative-serving      autoscaler-d795c8b8-wpncw                       0/1     ImagePullBackOff   0          14m
knative-serving      controller-799576cd4b-bgvsd                     0/1     ImagePullBackOff   0          14m
knative-serving      domain-mapping-546bd689f7-c2htx                 0/1     ImagePullBackOff   0          14m
knative-serving      domainmapping-webhook-5d9c67bc86-2nsbz          0/1     ImagePullBackOff   0          14m
knative-serving      webhook-6677f8d959-tklqm                        0/1     ImagePullBackOff   0          14m

```

Reply:
Some users might need more than 5 minutes on the first install since there is no cache, hard to know what's the correct number.

if we don't want to use for ever `-1` then make it much longer

Reply:
I've never had a fresh install take more than 3 minutes, so I think 5 minutes for this step should more than cover it. But if we want to be safe, I can bump it to 10. 

I do think we want to set _some_ timeout though, otherwise an imagePullBackOff (or a related error) would just cause the install to hang.

Reply:
I think 10m works